compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchTensorPass.cpp

```diff
@@ -0,0 +1,221 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <vector>
+#include "iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Patterns.h"
+#include "iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Utils.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowDialect.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/Flow/Transforms/ConvertRegionToWorkgroups.h"
+#include "iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/MemRef/Transforms/Transforms.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/Dominance.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Interfaces/FunctionInterfaces.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-flow-tensor-to-flow"
+
+namespace mlir::iree_compiler::IREE::Flow {
+
+#define GEN_PASS_DEF_DISPATCHTENSORPASS
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h.inc"
+
+/// Return `true` if the given op is contained in DispatchWorkgroupsOp or in a
+/// DispatchRegionOp.
+static bool isInDispatchRegion(Operation *op) {
+  return op->getParentOfType<IREE::Flow::DispatchWorkgroupsOp>() ||
+         op->getParentOfType<IREE::Flow::DispatchRegionOp>();
+}
+
+/// Wrap a single op in a DispatchWorkgroupsOp.
+static FailureOr<IREE::Flow::DispatchWorkgroupsOp>
+wrapInWorkgroupsOp(mlir::TensorDimTrackingRewriter &rewriter, Operation *op) {
+  // Simplify tensor::DimOps.
```

Comment:
nit: drop the comment. Because the code itself already document it (e.g., `xxx::simplifyDimOps`).